share/functions/fish_fossil_prompt.fish

```diff
@@ -4,48 +4,88 @@ function fish_fossil_prompt --description 'Write out the fossil prompt'
         return 1
     end
 
-    # Bail if not a fossil checkout
-    if not fossil ls &> /dev/null
-	return 127
-    end
+    # Read branch and bookmark (bail if not checkout)
+    set -l branch (fossil branch current 2>/dev/null)
+    or return 127
+
+    set -g fish_color_fossil_clean green
+    set -g fish_color_fossil_modified yellow
+    set -g fish_color_fossil_dirty red
+
+    set -g fish_color_fossil_added green
+    set -g fish_color_fossil_renamed magenta
+    set -g fish_color_fossil_missing red
+    set -g fish_color_fossil_deleted red
+    set -g fish_color_fossil_untracked yellow
+    set -g fish_color_fossil_conflict red
+
+    set -g fish_prompt_fossil_status_added '✚'
+    set -g fish_prompt_fossil_status_modified '*'
+    set -g fish_prompt_fossil_status_renamed '⇒'
+    set -g fish_prompt_fossil_status_deleted '-'
+    set -g fish_prompt_fossil_status_missing '✖'
+    set -g fish_prompt_fossil_status_untracked '?'
+    set -g fish_prompt_fossil_status_conflict '×'
+
+    set -g fish_prompt_fossil_status_order added modified renamed deleted missing untracked conflict
 
-    # Parse fossil info
-    set -l fossil_info (fossil info)
-    string match --regex --quiet \
-        '^project-name:\s*(?<fossil_project_name>.*)$' $fossil_info
-    string match --regex --quiet \
-        '^tags:\s*(?<fossil_tags>.*)$' $fossil_info
 
     echo -n ' ['
-    set_color --bold magenta
-    echo -n $fossil_project_name
-    set_color normal
-    echo -n ':'
-    set_color --bold yellow
-    echo -n $fossil_tags
-    set_color normal
+    # Disabling color and pager is always a good idea.
+    set -l repo_status (fossil changes --differ 2>/dev/null | grep -v '\w:\|^\s' | awk '{print $1}' | sort -u)
```

Comment:
This `grep` can be replaced with a `string match`. Possibly something like

```fish
string match -rv '\w:|^\s'
```

(but note I have no idea what the `\w:` is supposed to match and the dialects are slightly different - string is PCRE)

The awk would then probably be `string split " " -f1` and the `sort -u` can just be `path sort -u` (which is in `path` because it knows about sorting by directory or filename, it's perfectly cromulent for sorting ordinary strings excluding NULL-bytes).

Reply:
It would be helpful to have a sample of the output and what you want to get here.

Reply:
Seems good. Is path split slower than sort or is it just my luck with timing?
Output of the `status` command.
```
repository:   /tmp/aaa/foo.fossil
local-root:   /tmp/aaa/foo/
config-db:    /home/user/.fossil
checkout:     c30b371a740b617cc24f660791189c2b647cbcde 2023-02-02 07:26:59 UTC
parent:       57f300be2dbe273717dc10fd5941999ba5388907 2023-02-02 07:00:50 UTC
tags:         trunk
comment:      foo (user: user)
MISSING    a
EXTRA      b
EXTRA      c
ADDED      d
DELETED    e
RENAMED    b  ->  g
EXTRA      t
```

Reply:
I tried to hyperfine the script. Seems like the change is minor and I just had too much noise.